gluten-celeborn/velox/src/main/scala/org/apache/spark/shuffle/VeloxCelebornHashBasedColumnarShuffleWriter.scala

```diff
@@ -72,6 +73,12 @@ class VeloxCelebornHashBasedColumnarShuffleWriter[K, V](
       } else {
         val handle = ColumnarBatches.getNativeHandle(cb)
         if (nativeShuffleWriter == -1L) {
+          val startPartitionId = dep.nativePartitioning.getShortName match {
+            case "rr" =>
+              new XORShiftRandom(context.partitionId())
+                .nextInt(dep.partitioner.numPartitions)
+            case _ => 0
+          }
```

Comment:
@zhli1142015  thank you for the fix. Can we move this code block into NativePartitioning ? It should also be reused for other shuffle service.

Reply:
Moved this to shuffleUtils, thanks.